frontend/src/components/admin/UsersTable.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+
+import { Icon } from "@chakra-ui/icon";
+import { MdDelete } from "react-icons/md";
+import {
+  Badge,
+  Link,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  IconButton,
+} from "@chakra-ui/react";
+import { User } from "../../APIClients/queries/UserQueries";
+import convertLanguageTitleCase from "../../utils/LanguageUtils";
+
+export type UsersTableProps = {
+  isTranslators?: boolean;
+  users: User[];
+};
+
+const UsersTable = ({ isTranslators = false, users }: UsersTableProps) => {
+  const approvedLanguanges = users.map((userObj: User) => {
+    const approvedLanguange = isTranslators
+      ? userObj?.approvedLanguagesTranslation
+      : userObj?.approvedLanguagesReview;
+    return approvedLanguange
+      ? JSON.parse(approvedLanguange!!.trim().replace(/'/g, '"'))
+      : "";
+  });
+
+  const generateBadges = (approvedLanguages: any) =>
+    Object.keys(approvedLanguages).map((apprLang: string) => (
+      // TODO: determine badge color by label
+      <Badge background="orange.50">{`${convertLanguageTitleCase(
+        apprLang,
+      )} | Level ${approvedLanguages[apprLang]}`}</Badge>
+    ));
+
+  const tableBody = users.map((userObj: User, index: number) => (
+    <Tr>
+      <Td>
+        <Link
+          href={`/user/${userObj?.id}`}
+        >{`${userObj?.firstName} ${userObj?.lastName}`}</Link>
+      </Td>
+      <Td>
+        <Link href={`mailto:${userObj?.email}`}>{userObj?.email}</Link>
+      </Td>
+      <Td>{generateBadges(approvedLanguanges[index])}</Td>
+      {/* TODO: make delete button functional */}
+      <Td>
+        <IconButton
```

Comment:
the icon seems a bit off-centre from the heading (but that might just be me
![image](https://user-images.githubusercontent.com/32009013/137341478-d9e11360-0b3f-44a0-b3d0-896378c78e47.png)

Reply:
hmm I think I could only center it relative to the whole width of the cell, which is larger than the ACTION text, so it still wouldnt be centered. I'm fine with leaving it off to the left a bit - its technically aligned with the front of the ACTION column. LMK what you think!

Reply:
sounds good to me!